test/test-api-getreport.js

```diff
@@ -1,11 +1,11 @@
 'use strict';
 
-// Testcase for returning NodeReport as a string via API call
+// Testcase for returning report as a string via API call
 
 const common = require('./common.js');
 const tap = require('tap');
 const nodereport = require('../');
 var report_str = nodereport.getReport();
 common.validateContent(report_str, tap, {pid: process.pid,
-  commandline: [process.argv0, 'test/test-api-getreport.js'].join(' ')
-});
\ No newline at end of file
+  commandline: [process.argv[0], 'test/test-api-getreport.js'].join(' ')
```

Comment:
hmm so while everything passes when run with npm test, it looks like other ways of running the test fails:
1) If the test is manually run on Windows with backslashes, 
e.g.
```
node test\test-api-getreport.js
```
2) If the test is run manually anywhere other than the node-report root directory, 
e.g.
```
-bash-4.2$ cd test
-bash-4.2$ pwd
/home/users/riclau/sandbox/github/nodereport/test
-bash-4.2$ node test-api-getreport.js
TAP version 13
1..17
ok 1 - Checking report contains Node Report section
ok 2 - Checking report contains JavaScript Stack Trace section
ok 3 - Checking report contains JavaScript Heap section
ok 4 - Checking report contains System Information section
ok 5 - Node Report header section contains expected process ID
ok 6 - Node Report header section contains expected Node.js version
not ok 7 - Checking report contains expected command line
  ---
  found: >-
    Event: JavaScript API, location: "GetReport"

    Dump event time:  2017/02/14 18:35:19

    Module load time: 2017/02/14 18:35:19

    Process ID: 113131

    Command line: node test-api-getreport.js


    Node.js version: v4.7.3

    (ares: 1.10.1-DEV, http_parser: 2.7.0, icu: 56.1, modules: 46, openssl:
    1.0.2k,
     uv: 1.9.1, v8: 4.5.103.43, zlib: 1.2.8)

    node-report version: 2.0.0 (built against Node.js v4.7.3)


    OS version: Linux 3.10.0-327.18.2.el7.x86_64 #1 SMP Fri Apr 8 05:09:53 EDT
    2016

    (glibc: 2.17)


    Machine: drx-hemera x86_64
  pattern: >-
    /Command line: \/dev\/shm\/usenode.riclau\/node-v4.7.3-linux-x64\/bin\/node
    test\/test-api-getreport.js/
  at:
    line: 77
    column: 9
    file: common.js
    function: validateContent
  stack: |
    Object.validateContent (common.js:77:9)
    Object.<anonymous> (test-api-getreport.js:9:8)
  source: |
    t.match(nodeReportSection,
  ...
```
Also note that Node.js seems to fully qualify process.argv[0], so process.argv0 would be the correct thing to use if it is available (but would not fix the test looking for `test/test-api-getreport.js`).

3) If run with extra command line arguments, 
e.g.
```
-bash-4.2$ node --max-old-space-size=1024 test/test-api-getreport.js
TAP version 13
1..17
ok 1 - Checking report contains Node Report section
ok 2 - Checking report contains JavaScript Stack Trace section
ok 3 - Checking report contains JavaScript Heap section
ok 4 - Checking report contains System Information section
ok 5 - Node Report header section contains expected process ID
ok 6 - Node Report header section contains expected Node.js version
not ok 7 - Checking report contains expected command line
  ---
  found: >-
    Event: JavaScript API, location: "GetReport"

    Dump event time:  2017/02/14 18:43:24

    Module load time: 2017/02/14 18:43:24

    Process ID: 113385

    Command line: node --max-old-space-size=1024 test/test-api-getreport.js


    Node.js version: v4.7.3

    (ares: 1.10.1-DEV, http_parser: 2.7.0, icu: 56.1, modules: 46, openssl:
    1.0.2k,
     uv: 1.9.1, v8: 4.5.103.43, zlib: 1.2.8)

    node-report version: 2.0.0 (built against Node.js v4.7.3)


    OS version: Linux 3.10.0-327.18.2.el7.x86_64 #1 SMP Fri Apr 8 05:09:53 EDT
    2016

    (glibc: 2.17)


    Machine: drx-hemera x86_64
  pattern: >-
    /Command line: \/dev\/shm\/usenode.riclau\/node-v4.7.3-linux-x64\/bin\/node
    \/home\/users\/riclau\/sandbox\/github\/nodereport\/test\/test-api-getreport.js/
  at:
    line: 77
    column: 9
    file: test/common.js
    function: validateContent
  stack: |
    Object.validateContent (test/common.js:77:9)
    Object.<anonymous> (test/test-api-getreport.js:9:8)
  source: |
    t.match(nodeReportSection,
  ...
```


The other tests that check the expected command line are creating child processes to get the report from, so know exactly what command line is invoked. This test doesn't have that control, so is less robust if run in ways other than `npm test`. Are we concerned about these?